schema/component.schema.json

```diff
@@ -0,0 +1,52 @@
+{
+  "$anchor": "_hint-component",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "$patch": {
```

Comment:
I think this should be $merge for component schemas?

Reply:
https://adapt-security.github.io/adapt-authoring-documentation/manual/#/writing-a-schema?id=defining-schema-inheritance